naga/src/back/msl/mod.rs

```diff
@@ -221,7 +221,7 @@ impl Default for Options {
 }
 
 /// A subset of options that are meant to be changed per pipeline.
-#[derive(Debug, Default, Clone, PartialEq, Eq, Hash)]
```

Comment:
question: [You shouldn't have to remove `PartialEq, Eq` here on account of a `HashMap`.](https://doc.rust-lang.org/std/collections/struct.HashMap.html#impl-Eq-for-HashMap%3CK,+V,+S%3E) Is the intention of removing this now to be forwards-compatible with an arbitrary choice of map-like data structure down the road, or something?

This question also applies to other places where these `derive`s have been removed in like manner.

Reply:
I had to remove `Eq` (due to the `f64`) and `Hash` (due to the `HashMap`). I highly doubt anyone was relying on these anyway (they were most likely copy-pasted from the `Options` structs).

Reply:
D'oh, right, `f64` doesn't implement `Eq`. 👍🏻